app/controllers/redirections_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RedirectionsController < ApplicationController
+  def next
+    redirection = Redirection.find_by(slug: params[:slug])
```

Comment:
Yup, that's what I'm thinking. If we need to we can always deregister or blacklist or whitelist people.